### PR TITLE
Feature t8 element root

### PR DESCRIPTION
**Pull request by lukasdreyer**
**_Describe your changes here:_**
Introduce an element function that fills an element with the root element. 
Also replaces all occurences of `t8_element_set_linear_id (elem, 0, 0)` by `t8_element_root`.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [x] The reviewer executed the new code features at least once and checked the results manually

- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] New source/header files are properly added to the Makefiles
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [x] The code is covered in an existing or new test case using Google Test

#### Github action

- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [x] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [x] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [x] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [x] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: src/t8_schemes/t8_default/t8_default_prism/t8_default_prism_cxx.cxx [hunk 1]
```diff
@@ -476,5 +476,16 @@ t8_default_scheme_prism_c::~t8_default_scheme_prism_c ()
    * However we need to provide an implementation of the destructor
    * and hence this empty function. */
 }
+void
+t8_default_scheme_prism_c::t8_element_root (t8_element_t *elem) const
+{
+  t8_dprism_t *prism = (t8_dprism_t *) elem;
```
Comment (Davknapp) [inline src/t8_schemes/t8_default/t8_default_prism/t8_default_prism_cxx.cxx:482]: I think it is better to call the root functions of the tri and the line here.

File: src/t8_schemes/t8_default/t8_default_tet/t8_default_tet_cxx.cxx [hunk 2]
```diff
@@ -564,5 +564,14 @@ t8_default_scheme_tet_c::~t8_default_scheme_tet_c ()
    * However we need to provide an implementation of the destructor
    * and hence this empty function. */
 }
-
+void
```
Comment (Davknapp) [inline src/t8_schemes/t8_default/t8_default_tet/t8_default_tet_cxx.cxx:567]: There is no need to implement it here, if you use the `T8_DTRI_TO_DTET` logic in the triangle implementation

File: src/t8_schemes/t8_default/t8_default_tri/t8_default_tri_cxx.cxx [hunk 3]
```diff
@@ -573,5 +573,13 @@ t8_default_scheme_tri_c::~t8_default_scheme_tri_c ()
...
    * and hence this empty function. */
 }
-
+void
+t8_default_scheme_tri_c::t8_element_root (t8_element_t *elem) const
+{
+  t8_dtri_t *tri = (t8_dtri_t *) elem;
+  tri->level = 0;
+  tri->x = 0;
+  tri->y = 0;
```
Comment (Davknapp) [inline src/t8_schemes/t8_default/t8_default_tri/t8_default_tri_cxx.cxx:582]: you can use the `T8_DTRI_TO_DTET` logic here to implement both roots at once.

File: test/t8_schemes/t8_gtest_root.cxx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2015 the developers
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_root.cxx:6]:
```suggestion
Copyright (C) 2024 the developers
```
Using 'Configure User Snippets` in VsCode you can implement your own snippet to fill in the license with the current year ;)

File: test/t8_schemes/t8_gtest_root.cxx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  t8_eclass_scheme_c *ts;
+  t8_eclass_t eclass;
+};
+
+/*Test root*/
+
+TEST_P (root, has_level_zero)
+{
+  EXPECT_EQ (ts->t8_element_level (element), 0);
+}
```
Comment (Davknapp) [inline test/t8_schemes/t8_gtest_root.cxx:59]: we still expect that root equals t8_element_linear_id(0,0), right? Can you implement a test for that too?

**Review by Davknapp — CHANGES_REQUESTED** [review]

File: src/t8_schemes/t8_default/t8_default_prism/t8_default_prism_cxx.cxx (hunk 1 above)
Comment (lukasdreyer) [reply src/t8_schemes/t8_default/t8_default_prism/t8_default_prism_cxx.cxx:482]: I didn't want to touch 16 additional files, therefore I didn't put the implementation in the `*_bits.c` file. Thus, I cannot call the tri and line functionality here. The same applies for your comment regarding `T8_DTRI_TO_DTET` as I do not have the base functionality which could be compiled once with and once without `T8_DTRI_TO_DTET`. 

I will do it if that is our policy, but I am not completely sure of the benefits to have everything in the `*_bits.c`

Comment (Davknapp) [reply src/t8_schemes/t8_default/t8_default_prism/t8_default_prism_cxx.cxx:482]: Ah yes, you are right, I missed that. That is ok for me.

**Review by Davknapp — APPROVED** [review]

**Pull request merged**